README.md

```diff
@@ -1,78 +1,268 @@
+*Request format:*
 
-You may want to choose a plugin name rather than using the default one. To do so, edit the `manifest.json` file at the root of this repo and change the `name` property to your convenience.
+```javascript
+{
+  // Kuzzle API params
+  "controller": "s3/file",
+  "action": "delete",
 
+  "fileKey": "xen/<uuid>/headcrab.png" // File key in S3 bucket
+}
+```
 
-## `manifest.json` file
+## Configuration
 
-The `manifest.json` file is here to provide a description of your plugin to Kuzzle:
+You need to set your AWS access key: `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.  
+
+In your `kuzzlerc` file, you can change the following configuration variable:
 
 ```js
 {
-  /**
-   * This is metadata to describe your plugin
-   */
-  "name": "name-of-your-plugin",
-
-  /**
-   * Define what Kuzzle version this plugin is designed for.
-   * Use semver notation to limit the range of supported Kuzzle versions.
-   * If not set, Kuzzle will complain and consider that the plugin has been 
-   * designed for Kuzzle v1 only.
-   */
-  "kuzzleVersion": ">=1.1.0 <2.0.0"
+  "plugins": {
+    "s3": {
+      // AWS S3 bucket
+      "bucketName": "your-s3-bucket",
+      // AWS S3 region
+      "region": "eu-west-3",
+      // TTL in ms before Presigned URL expire or the uploaded file is deleted
+      "signedUrlTTL": 1200000,
+      // Redis key prefix
+      "redisPrefix": "s3Plugin/uploads"
+    }
+  }
 }
 ```
+
+
+### AWS S3 Bucket
+
+First you must configure your bucket to allow public access to uploaded files.  
+Go to the `Permissions` tab in your bucket configuration and in `Bucket Policy` add the following statement:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "AddPerm",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "s3:GetObject",
+      "Resource": "arn:aws:s3:::your-bucket-name/*"
+    }
+  ]
+}
+```
+
+Then you have to allow Cross Origin Request by editing the CORS Configuration:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<CORSConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+<CORSRule>
+    <AllowedOrigin>your-app-domain.com</AllowedOrigin>
+    <AllowedMethod>GET</AllowedMethod>
+    <AllowedMethod>PUT</AllowedMethod>
+    <AllowedMethod>POST</AllowedMethod>
+    <MaxAgeSeconds>3000</MaxAgeSeconds>
+    <AllowedHeader>Content-Type</AllowedHeader>
+    <AllowedHeader>Authorization</AllowedHeader>
+</CORSRule>
+</CORSConfiguration>
+```
+
+## Installation
+
+### Node.js 8 and beyond
+
+Clone this repository in your `plugins/available` directory and then link it to the `plugins/enabled` directory.  
+
+For more information, refer to the official documentation: https://docs.kuzzle.io/guide/1/essentials/plugins/#installing-a-plugin
+
+### Node.js 6
+
+Clone this repository in your `plugins/available` directory and then link it to the `plugins/enabled` directory.  
+
+Then go to your plugin directory and run `npm run es5-compat`.  
```

Comment:
(open to debate)

I wonder if it wouldn't be easier/better to have a global `index.Js` entrypoint trying to load the built compatibility version and, if it isn't available, fallbacking on the ES2018 version?

That would mean having a build script, doing nothing if `process.versions.node` is version 8 or newer (using [semver](https://www.npmjs.com/package/semver) for comparison?)

Reply:
I like this solution !